src/main/java/com/gameshop/ecommerce/user/service/MyAccountService.java

```diff
@@ -0,0 +1,115 @@
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        return userBuilder(user);
+    }
+
+    public void deleteInfo(UUID userId){
+        localUserDAO.deleteById(userId);
+    }
+
+    public LocalUserDto updateInfo(UUID userId, LocalUserDto userDto){
+        LocalUser user = localUserDAO.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        if(!user.getFirstName().equals(userDto.getFirstName())||
+                !user.getLastName().equals(userDto.getLastName())||
+                !user.getPhoneNumber().equals(userDto.getPhoneNumber())||
+                !user.getEmail().equals(userDto.getEmail())) {
+            if(userDto.getFirstName() != null){
+                user.setFirstName(userDto.getFirstName());
+            }else {
+                user.setFirstName(user.getFirstName());
+            }
+            if(userDto.getLastName() != null){
+                user.setLastName(userDto.getLastName());
+            }else{
+                user.setLastName(user.getLastName());
+            }
+            if(userDto.getPhoneNumber() != null){
+                user.setPhoneNumber(userDto.getPhoneNumber());
+            }else{
+                user.setPhoneNumber(user.getPhoneNumber());
+            }
+            if(userDto.getEmail() != null){
+                user.setEmail(userDto.getEmail());
+            }else{
+                user.setEmail(user.getEmail());
+            }
+            if(userDto.getNewPassword() == null){
+                LocalUser updatedUser = localUserDAO.save(user);
+                log.info("Information about user was updated successfully");
+                return localUserMapper.entityToDto(updatedUser);
+            } else{
+                return updatePassword(user,userDto);
+            }
+
+        }
+        else{
+            user.setFirstName(user.getFirstName());
+            user.setLastName(user.getLastName());
+            user.setPhoneNumber(user.getPhoneNumber());
+            user.setEmail(user.getEmail());
+            if(userDto.getNewPassword() == null){
+                LocalUser updatedUser = localUserDAO.save(user);
+                return localUserMapper.entityToDto(updatedUser);
+            }else{
+                log.info("No new information. Checking password");
+                return updatePassword(user,userDto);
+            }
+        }
+    }
+
+    public LocalUserDto updatePassword(LocalUser user, LocalUserDto userDto){
+        if(encryptionService.checkPassword(userDto.getOldPassword(), user.getPassword())){
+            user.setPassword(encryptionService.encryptPassword(userDto.getNewPassword()));
+            LocalUser updatedUser = localUserDAO.save(user);
+            return localUserMapper.entityToDto(updatedUser);
+        }else{
+            log.info("Old password is incorrect");
+            LocalUser updatedUser = localUserDAO.save(user);
```

Comment:
If user want change password, but send incorrect old pw, we need inform about it